theano/tensor/opt.py

```diff
@@ -5841,3 +5841,26 @@ def local_add_mul_fusion(node):
 def local_grad_clip(node):
     if isinstance(node.op, theano.gradient.GradClip):
         return node.inputs
+
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@gof.local_optimizer([T.Alloc])
+def local_reduce_double_alloc(node):
```

Comment:
The name is already taken

On Mon, Jun 29, 2015 at 3:27 PM, Frédéric Bastien <notifications@github.com>
wrote:

> In theano/tensor/opt.py
> <https://github.com/Theano/Theano/pull/3082#discussion_r33502102>:
>
> > @@ -5841,3 +5841,26 @@ def local_add_mul_fusion(node):
> >  def local_grad_clip(node):
> >      if isinstance(node.op, theano.gradient.GradClip):
> >          return node.inputs
> > +
> > +@register_canonicalize
> > +@register_stabilize
> > +@register_specialize
> > +@gof.local_optimizer([T.Alloc])
> > +def local_reduce_double_alloc(node):
>
> Rename local_useless_alloc
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3082/files#r33502102>.
>